### PR TITLE
Maven artifacts relases into Apache

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Added configuration to publish on ASF Nexus Maven repository for both snapshot and releases.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -927,15 +873,4 @@ flexible messaging model and an intuitive client API.</description>
       <url>http://yahoo.bintray.com/maven</url>
     </repository>
   </repositories>
-
-  <distributionManagement>
-    <snapshotRepository>
```
Comment (rdhabalia) [inline pom.xml]: should we add `Apache Nexus repo url` in `distributionManagement`.? How would travis know the repo url? does `Papache-release` indicates repo-url?

File: .travis.yml [hunk 2]
```diff
@@ -56,19 +56,16 @@ script:
...
+     provider: script
+     skip_cleanup: true
+     script: mvn deploy -DskipTests --settings .travis/settings.xml
+     on:
+       tags: false
   -
     provider: script
     skip_cleanup: true
-    script: mvn install deploy -Prelease -DskipTests --settings .travis/settings.xml
+    script: mvn install deploy -Papache-release -DskipTests --settings .travis/settings.xml
```
Comment (rdhabalia) [inline .travis.yml]: should we add `-DcreateChecksum=true` to create checksum with it?

File: pom.xml (hunk 1 above)
Comment (merlimat) [inline pom.xml]: The parent pom is set to `org.apache:apache` and from that we're inheriting all the right settings.

I have followed the instructions at http://www.apache.org/dev/publishing-maven-artifacts.html

File: .travis.yml (hunk 2 above)
Comment (merlimat) [inline .travis.yml]: Just tried with the current config: 

```shell
export APACHE_USER=XYZ
export APACHE_PASSWORD=123
mvn deploy -DskipTests --settings .travis/settings.xml
```

Uploaded snapshot artifacts are already available at:
https://repository.apache.org/content/repositories/snapshots/org/apache/pulsar/pulsar/1.19-incubating-SNAPSHOT/

Comment (merlimat) [inline .travis.yml]: Checksum are already created as well (md5 and sha1). The pgp signature is only done for releases.

**Pull request merged**